src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/SynthesizedClosureEnvironment.cs

```diff
@@ -126,6 +126,11 @@ internal override IEnumerable<FieldSymbol> GetFieldsToEmit()
 
         public override Symbol ContainingSymbol => _topLevelMethod.ContainingSymbol;
 
+        // Closures in the same method share the same SynthesizedClosureEnvironment. We must
+        // always return true because two closures in the same method might have different
```

Comment:
>always return true because two closures in the same method might have different [](start = 11, length = 79)

I am not sure why this matters. Doesn't SynthesizedClosureEnvironment represent a single closure?

Reply:
No, we might stick multiple closures in the same ClosureEnvironment if they share the same captured variables.